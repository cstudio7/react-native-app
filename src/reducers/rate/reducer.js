import {
  RATE_APP,
  UPDATE_USER_VISITS,
  RESET_USER_VISITS
} from '../../constants/ActionTypes';
import createReducer from '../../modules/createReducer/createReducer';

const initialState = {
  visitsCount: 0,
  rated: false
};

const rate = state => {
  return {
    ...state,
    rated: true,
    visitsCount: 0
  };
};

const updateUserVisits = state => {
  return {
    ...state,
    visitsCount: state.visitsCount + 1
  };
};

const resetUserVisits = state => {
  return {
    ...state,
    visitsCount: 0
  };
};

export default createReducer(initialState, {
  [RATE_APP]: rate,
  [UPDATE_USER_VISITS]: updateUserVisits,
  [RESET_USER_VISITS]: resetUserVisits
});
