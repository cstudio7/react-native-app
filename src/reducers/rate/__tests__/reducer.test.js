import rate from '../reducer';
import {
  RATE_APP,
  UPDATE_USER_VISITS,
  RESET_USER_VISITS
} from '../../../constants/ActionTypes';

it('returns the initial state', () => {
  expect(rate(undefined, {})).toMatchSnapshot();
});

it('update user visits counter', () => {
  const initialState = {
    rated: true,
    visitsCount: 0
  };

  expect(
    rate(initialState, {
      type: UPDATE_USER_VISITS
    })
  ).toEqual({
    rated: true,
    visitsCount: 1
  })
});

it('resets user visits', () => {
  const initialState = {
    rated: false,
    visitsCount: 3
  };

  expect(
    rate(initialState, {
      type: RESET_USER_VISITS
    })
  ).toEqual({
    rated: false,
    visitsCount: 0
  })
});

it('handles RATE_APP action and marks app as rated', () => {
  const initialState = {
    rated: false,
    visitsCount: 0
  };

  expect(
    rate(initialState, {
      type: RATE_APP
    })
  ).toMatchSnapshot();
});
