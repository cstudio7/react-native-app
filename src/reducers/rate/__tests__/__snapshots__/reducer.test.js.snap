// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles RATE_APP action and marks app as rated 1`] = `
Object {
  "rated": true,
  "visitsCount": 0,
}
`;

exports[`returns the initial state 1`] = `
Object {
  "rated": false,
  "visitsCount": 0,
}
`;
